kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,10 +1327,10 @@ class ResidualBasedBlockBuilderAndSolver
                 }
 
                 // Merging all the temporal indexes
-                for (int i = 0; i < static_cast<int>(temp_indices.size()); ++i) {
-                    lock_array[i].lock();
-                    indices[i].insert(temp_indices[i].begin(), temp_indices[i].end());
-                    lock_array[i].unlock();
+                for (auto& pair_temp_indices : temp_indices) {
+                    lock_array[pair_temp_indices.first].lock();
+                    indices[pair_temp_indices.first].insert(pair_temp_indices.second.begin(), pair_temp_indices.second.end());
+                    lock_array[pair_temp_indices.first].unlock();

```

Comment:
I don't get it. Why is this in a naked parallel region in the first place?

Reply:
actually, no I don't get it. For some reason it seems to me that we're computing the exact same thing on all threads, and the only reason why we're not getting duplicate results is because `set`s are being used.

Reply:
In fact, it is because we are using the locks

Reply:
aight, @sunethwarna explained what's happening here